### ISSUE TITLE
1.11.4 - CoilCoolingDXTwoStageWithHumidityControlMode missing required fields

### ISSUE THREAD

**Issue by MatthewSteen** [NREL/OpenStudio#2253]
Causes the sim to fail. 

**in.idf**

![image](https://cloud.githubusercontent.com/assets/8397034/15755680/1c80a308-28bb-11e6-9dcc-5e26d052697a.png)

**Comment by MatthewSteen** [comment]
When dropping a Coil:Cooling:DX:TwoStageWithHumidityControlMode into an AirLoopHVAC the following required fields are missing after forward translation causing the sim to fail.

* Normal Mode Stage 1 Coil Performance Object Type
* Normal Mode Stage 1 Coil Performance Name

**OSM**
```
OS:Coil:Cooling:DX:TwoStageWithHumidityControlMode,
  {29763528-d754-447b-9739-899819872944}, !- Handle
  Two Stage DX Coil With Humidity Control, !- Name
  ,                                       !- Availability Schedule
  {c9e383f7-1103-43f1-8dfd-be21d6dee5bb}, !- Air Inlet Node Name
  {694f74ee-39cb-4067-9f08-9f12745506fc}, !- Air Outlet Node Name
  0,                                      !- Crankcase Heater Capacity {W}
  10,                                     !- Maximum Outdoor Dry-Bulb Temperature for Crankcase Heater Operation {C}
  2,                                      !- Number of Capacity Stages
  1,                                      !- Number of Enhanced Dehumidification Modes
  ,                                       !- Normal Mode Stage 1 Coil Performance
  ,                                       !- Normal Mode Stage 1 Plus 2 Coil Performance
  ,                                       !- Dehumidification Mode 1 Stage 1 Coil Performance
  ,                                       !- Dehumidification Mode 1 Stage 1 Plus 2 Coil Performance
  ,                                       !- Supply Water Storage Tank
  ,                                       !- Condensate Collection Water Storage Tank
  0,                                      !- Basin Heater Capacity {W/K}
  2,                                      !- Basin Heater Setpoint Temperature {C}
  ;                                       !- Basin Heater Operating Schedule
```

**IDF**
```
Coil:Cooling:DX:TwoStageWithHumidityControlMode,
  Two Stage DX Coil With Humidity Control, !- Name
  ,                                       !- Availability Schedule Name
  Node 93,                                !- Air Inlet Node Name
  Node 90,                                !- Air Outlet Node Name
  0,                                      !- Crankcase Heater Capacity {W}
  10,                                     !- Maximum Outdoor Dry-Bulb Temperature for Crankcase Heater Operation {C}
  2,                                      !- Number of Capacity Stages
  1,                                      !- Number of Enhanced Dehumidification Modes
  ,                                       !- Normal Mode Stage 1 Coil Performance Object Type
  ,                                       !- Normal Mode Stage 1 Coil Performance Name
  ,                                       !- Normal Mode Stage 1+2 Coil Performance Object Type
  ,                                       !- Normal Mode Stage 1+2 Coil Performance Name
  ,                                       !- Dehumidification Mode 1 Stage 1 Coil Performance Object Type
  ,                                       !- Dehumidification Mode 1 Stage 1 Coil Performance Name
  ,                                       !- Dehumidification Mode 1 Stage 1+2 Coil Performance Object Type
  ,                                       !- Dehumidification Mode 1 Stage 1+2 Coil Performance Name
  ,                                       !- Supply Water Storage Tank Name
  ,                                       !- Condensate Collection Water Storage Tank Name
  0,                                      !- Basin Heater Capacity {W/K}
  2;                                      !- Basin Heater Setpoint Temperature {C}
```

**Comment by MatthewSteen** [comment]
#2313

**Comment by eringold** [comment]
FYI, still valid in v2.2.0.

**Comment by eringold** [comment]
Workaround: live dangerously; copy coils and curves manually from `hvac_library.osm`

**Comment by jmarrec** [comment]
Fixed in  #2825.

The problem was that the clone method doesn't specifically handle the `CoilPerformanceDXCooling` objects, and if you clone in another model, they don't come with, causing a fatal E+ crash (two out of four are actually required-fields)
`CoilPerformanceDXCooling` is a `ParentObject` because it has curves. But it is also only used by one object: the `CoilCoolingDXTwoStageWithHumidityControl`. As such it is also a resources. I've tried several ways, but none worked properly. So in the end I've left it as a ParentObject, and I'm explicitly handling it in `CoilCoolingDXTwoStageWithHumidityControl::clone` method.

The new behavior is as follows:
* If you clone in the same model, the coil points to the same CoilPerformanceDXCooling objects as the original (so treating it as a "resource" effectively).
* If you clone in another model, they are also cloned.

I added a Model GTest to test this behavior.
